eng/SourceBuildPrebuiltBaseline.xml

```diff
@@ -28,5 +28,11 @@
     <!-- Analyzers cannot be SBRPed + an older release version is being used, meaning there is no intermediate for it
          In full source build, this version is overridden by previous repo outputs or come from previously source-built artifacts. -->
     <UsagePattern IdentityGlob="Microsoft.CodeAnalysis.Analyzers/*" />
+
+    <!-- Allowed due to https://github.com/dotnet/source-build-reference-packages/pull/979 -->
```

Comment:
I suspect these packages are lifted during the product source-build therefore these versions are only being used during the repo's source-build leg.  When working on https://github.com/dotnet/source-build-reference-packages/pull/979, I ran a full product source build to ensure none of these were required.

Per https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages

> Note: Reference packages should only be added to this repo if they are required during the product source build (e.g. a [VMR](https://github.com/dotnet/dotnet) build). Reference packages that are only required for building a repo level source build should not be added to this repo. In this case, it is appropriate to add these types of package as allowed prebuilt via the eng/SourceBuildPrebuiltBaseline.xml file. See the [Eliminating pre-builts documentation](https://github.com/dotnet/source-build/blob/main/Documentation/eliminating-pre-builts.md) for detailed guidance.

Therefore it is correct to include these as allowed prebuilts but the reference to https://github.com/dotnet/source-build-reference-packages/pull/979 is not the correct reason they are allowed here as previously mentioned.

Reply:
> I see https://github.com/dotnet/aspnetcore/pull/56771 is the plan really to have each repo do this? It feels wrong to have SB bookkeeping break CI from flow. I've turned off batching for sbrp in main.

@MichaelSimons I think @lewing is bringing up a valid point. Removing SBRP packages that aren't used in the full product build but are in the repo build will break dependency flow. Is there a way to better rollout or stage these changes?

Reply:
I am open to suggestions on how to rollout these changes better.  I try to limit this "cleanup" to just before we enter RC.  I don't expect each repo will be affected by this change, it will only be a small subset.